db/migrate/20171225012500_create_question_text_response_comprehension.rb

```diff
@@ -0,0 +1,45 @@
+class CreateQuestionTextResponseComprehension < ActiveRecord::Migration[5.0]
+  def change
+    add_column :course_assessment_question_text_responses, :is_comprehension, :boolean, default: false
+
+    create_table :course_assessment_question_text_response_compre_groups do |t|
+      t.references :question,
+                   null: false,
+                   index: {
+                     name: :fk__course_assessment_text_response_compre_group_question
+                   },
+                   foreign_key: {
+                     to_table: :course_assessment_question_text_responses
+                   }
+      t.decimal :maximum_group_grade, precision: 4, scale: 1, null: false, default: 0.0
+    end
+
+    create_table :course_assessment_question_text_response_compre_points do |t|
+      t.references :group,
+                   null: false,
+                   index: {
+                     name: :fk__course_assessment_text_response_compre_point_group
+                   },
+                   foreign_key: {
+                     to_table: :course_assessment_question_text_response_compre_groups
+                   }
+      t.decimal :maximum_point_grade, precision: 4, scale: 1, null: false, default: 0.0
+    end
+
+    create_table :course_assessment_question_text_response_compre_solutions do |t|
+      t.references :point,
+                   null: false,
+                   index: {
+                     name: :fk__course_assessment_text_response_compre_solution_point
+                   },
+                   foreign_key: {
+                     to_table: :course_assessment_question_text_response_compre_points
+                   }
+      t.integer :solution_type, null: false, default: 0
+      t.text :solution, array: true, null: false, default: []
```

Comment:
Can you give some examples of `solution` and `solution_lemma`? Can I say:
 - If `solution` has `['sorting', 'classify', 'separation']`, then
 - `solution_lemma` has `['sort', 'classify', 'separate']`?

Reply:
Yup, you are right. The words in `solution_lemma` is the result of the lemmatiser.

Reply:
@makqien have a question. Is it important to store the non-lemma form of the solution?

Reply:
Yes. Because the non-lemma form will be shown to instructors when they edit the question. The lemma forms are hidden from view.

Reply:
So if the staff changes 1 word, you will concatenate the entire solution into 1 string and make an API call to get the new lemma forms?

Reply:
Yes, that would be the plan for now.